src/sentry/snuba/metrics.py

```diff
@@ -196,7 +205,12 @@ def _parse_orderby(self, query_params):
 
         return (op, metric_name), direction
 
-    def _parse_limit(self, query_params):
+    def _parse_limit(self, query_params, paginator_kwargs):
+        if self.orderby:
+            paginator_limit = paginator_kwargs.get("limit")
+            if paginator_limit is not None:
+                return paginator_limit
+
```

Comment:
nit: It might make sense to move the `per_page` validation here as well (from `_parse_orderby`), such that it's all in one place.